test/parallel/test-buffer-alloc.js

```diff
@@ -1147,6 +1155,22 @@ assert.throws(() => {
   assert.strictEqual(ubuf.buffer.byteLength, 10);
 }
 
+// Invalid length of Buffer.utf8Write
+{
+  const ubuf = Buffer.allocUnsafeSlow(2 ** 32);
+  assert.throws(() => {
+    ubuf.utf8Write('a', 2, kStringMaxLength + 2);
+  }, stringTooLongError);
+}
+
+// Invalid length of Buffer.write
+{
+  const ubuf = Buffer.allocUnsafeSlow(2 ** 32);
+  assert.throws(() => {
+    ubuf.write('a', 2, kStringMaxLength + 2);
+  }, stringTooLongError);
+}
+
```

Comment:
This test probably won't work in `test/parallel`, since 2 GiB is a lot to allocate (maybe it'll work on Linux)? It might work to put it in `test/sequential` instead.

Reply:
thanks @kvakil

Reply:
thanks your suggestion @joyeecheung , I have update it, add a method in `StringBytes` to ensure that the capacity of buffer remains within the integer range when operating with `string`